File: src/main/java/com/itsci/mju/maebanjumpen/controller/HousekeeperController.java
package com.itsci.mju.maebanjumpen.controller;

import com.itsci.mju.maebanjumpen.dto.HousekeeperDTO;
import com.itsci.mju.maebanjumpen.dto.HousekeeperDetailDTO; // 💡 เพิ่ม import DTO ตัวใหม่
import com.itsci.mju.maebanjumpen.service.HousekeeperService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/maeban/housekeepers")
public class HousekeeperController {

    @Autowired
    private HousekeeperService housekeeperService;

    // ... (เมธอด getAllHousekeepers เหมือนเดิม)
    @GetMapping
    public ResponseEntity<List<HousekeeperDTO>> getAllHousekeepers() {
        List<HousekeeperDTO> housekeepers = housekeeperService.getAllHousekeepers();
        return ResponseEntity.ok(housekeepers);
    }

    // 🎯 การแก้ไข 1: เปลี่ยนชื่อ Endpoint และชนิดข้อมูลที่ส่งคืน
    // Endpoint นี้ใช้สำหรับดึง "รายละเอียด" ที่มี JobsCompleted และ Reviews
    @GetMapping("/{id}")
    public ResponseEntity<HousekeeperDetailDTO> getHousekeeperDetailById(@PathVariable int id) {
        // ใช้เมธอด getHousekeeperDetailById จาก Service
        HousekeeperDetailDTO housekeeper = housekeeperService.getHousekeeperDetailById(id);
        if (housekeeper == null) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.ok(housekeeper);
    }

    // ... (เมธอด getHousekeepersByStatus เหมือนเดิม)
    @GetMapping("/status/{status}")
    public ResponseEntity<List<HousekeeperDTO>> getHousekeepersByStatus(@PathVariable String status) {
        List<HousekeeperDTO> housekeepers = housekeeperService.getHousekeepersByStatus(status);
        return ResponseEntity.ok(housekeepers);
    }

    // ... (เมธอด getUnverifiedOrNullStatusHousekeepers เหมือนเดิม)
    @GetMapping("/unverified-or-null")
    public ResponseEntity<List<HousekeeperDTO>> getUnverifiedOrNullStatusHousekeepers() {
        List<HousekeeperDTO> housekeepers = housekeeperService.getNotVerifiedOrNullStatusHousekeepers();
        return ResponseEntity.ok(housekeepers);
    }

    // ... (เมธอด CRUD อื่นๆ เหมือนเดิม)
    @PostMapping
    public ResponseEntity<HousekeeperDTO> createHousekeeper(@RequestBody HousekeeperDTO housekeeper) {
        HousekeeperDTO savedHousekeeper = housekeeperService.saveHousekeeper(housekeeper);
        return ResponseEntity.ok(savedHousekeeper);
    }

    @PutMapping("/{id}")
    public ResponseEntity<HousekeeperDTO> updateHousekeeper(@PathVariable int id, @RequestBody HousekeeperDTO housekeeper) {
        HousekeeperDTO updatedHousekeeper = housekeeperService.updateHousekeeper(id, housekeeper);
        if (updatedHousekeeper == null) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.ok(updatedHousekeeper);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteHousekeeper(@PathVariable int id) {
        housekeeperService.deleteHousekeeper(id);
        return ResponseEntity.noContent().build();
    }

    // 💡 หากต้องการแยก Endpoint ดึง HousekeeperDTO (ข้อมูลพื้นฐาน)
    // ควรสร้างเมธอดใหม่ เช่น getHousekeeperBaseById และเพิ่มใน Service
    // แต่สำหรับตอนนี้ การใช้ getHousekeeperDetailById ใน /housekeepers/{id} ก็เพียงพอแล้ว
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/housekeeperskill/controller/HousekeeperController.kt
package com.itsci.mju.maebanjumpen.housekeeperskill.controller

import com.itsci.mju.maebanjumpen.housekeeperskill.dto.HousekeeperDetailDTO
import com.itsci.mju.maebanjumpen.partyrole.dto.HousekeeperDTO
import com.itsci.mju.maebanjumpen.partyrole.service.HousekeeperService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/maeban/housekeepers")
class HousekeeperController(private val housekeeperService: HousekeeperService) {

    @GetMapping
    fun getAllHousekeepers(): ResponseEntity<List<HousekeeperDTO>> {
        val housekeepers = housekeeperService.getAllHousekeepers()
        return ResponseEntity.ok(housekeepers)
    }

    @GetMapping("/{id}")
    fun getHousekeeperDetailById(@PathVariable id: Int): ResponseEntity<HousekeeperDetailDTO> {
        val housekeeper = housekeeperService.getHousekeeperDetailById(id)
            ?: return ResponseEntity.notFound().build()
        return ResponseEntity.ok(housekeeper)
    }

    @GetMapping("/status/{status}")
    fun getHousekeepersByStatus(@PathVariable status: String): ResponseEntity<List<HousekeeperDTO>> {
        val housekeepers = housekeeperService.getHousekeepersByStatus(status)
        return ResponseEntity.ok(housekeepers)
    }

    @GetMapping("/unverified-or-null")
    fun getUnverifiedOrNullStatusHousekeepers(): ResponseEntity<List<HousekeeperDTO>> {
        val housekeepers = housekeeperService.getNotVerifiedOrNullStatusHousekeepers()
        return ResponseEntity.ok(housekeepers)
    }

    @PostMapping
    fun createHousekeeper(@RequestBody housekeeper: HousekeeperDTO): ResponseEntity<HousekeeperDTO> {
        val savedHousekeeper = housekeeperService.saveHousekeeper(housekeeper)
        return ResponseEntity.ok(savedHousekeeper)
    }

    @PutMapping("/{id}")
    fun updateHousekeeper(@PathVariable id: Int, @RequestBody housekeeper: HousekeeperDTO): ResponseEntity<HousekeeperDTO> {
        val updatedHousekeeper = housekeeperService.updateHousekeeper(id, housekeeper)
            ?: return ResponseEntity.notFound().build()
        return ResponseEntity.ok(updatedHousekeeper)
    }

    @DeleteMapping("/{id}")
    fun deleteHousekeeper(@PathVariable id: Int): ResponseEntity<Void> {
        housekeeperService.deleteHousekeeper(id)
        return ResponseEntity.noContent().build()
    }
}

